app/views/welcome/index.html.erb

```diff
@@ -1,4 +1,16 @@
 <h2>Welcome</h2>
 <p>
   The time is now: <%= Time.now %>
-</p>
\ No newline at end of file
+</p>
+
+<%= link_to 'View Topics', topics_url %>
+</br>
+<% if user_signed_in? %>
+  <%= link_to 'Log Out', destroy_user_session_url, method: :delete %>
+  <% else %>
+  <%= link_to 'Sign Up', new_user_registration_url %>
+  </br>
+  <%= link_to 'Log In', new_user_session_url %>
```

Comment:
I think the 'Log In' link should only show when the user is not signed in, so you need an else statement.